launch_ros/launch_ros/actions/lifecycle_transition.py

```diff
@@ -0,0 +1,257 @@
+        """
+        super().__init__(**kwargs)
+        self.__lifecycle_node_names = [
+            normalize_to_list_of_substitutions(name)
+            for name in lifecycle_node_names]
+        temp_transitions_ids = []
+        for id in transitions_ids:
+            temp_id = None
+            if isinstance(id, int):
+                temp_id = str(id)
+            else:
+                temp_id = id
+            temp_transitions_ids.append(temp_id)
+
+        self.__transition_ids = [
+            normalize_to_list_of_substitutions(id)
+            for id in temp_transitions_ids]
+
+        self.__event_handlers = {}
+        self.__logger = launch.logging.get_logger(__name__)
+
+    @classmethod
+    def parse(cls, entity: Entity, parser: Parser):
+        """Parse load_composable_node."""
+        _, kwargs = super().parse(entity, parser)
+
+        kwargs['lifecycle_node_names'] = parser.parser_substitution(
+            entity.get_attr('lifecycle_node_names', data_type=List[str])
+        )
+        # Probably need to make this parsable through a dict for convenience
+        kwargs['transitions_ids'] = entity.get_attr(
+            'transitions_ids', data_type=List[int])
+        return cls, kwargs
+
+    def _remove_event_handlers(
+            self,
+            context: LaunchContext,
+            node_name: str = None,
+            reason: str = None):
+        """Remove all consequent transitions if error occurs."""
+        if node_name is not None:
+            if reason is not None:
+                self.__logger.info("Stopping transitions for {} because {}.".format(
+                    node_name,
+                    reason
+                ))
+            for event_handler in self.__event_handlers[node_name]:
+                # Unregister event handlers and ignore failures, as these are
+                # already unregistered event handlers.
+                try:
+                    context.unregister_event_handler(event_handler=event_handler)
+                except ValueError:
+                    pass
+                finally:
+                    pass
+
+    def execute(
+        self,
+        context: launch.LaunchContext
+    ) -> Optional[List[Action]]:
+        """
+        Execute the Lifecycle Transition action.
+
+        :return Returns a list of actions to be executed to achieve specified transitions.
+          These are EventHandlers and EventEmitters for ChangeState and
+          StateTransition events of the nodes indicated.
+        """
+        lifecycle_node_names = [
+            perform_substitutions(context, name)
+            for name in self.__lifecycle_node_names]
+        subs_transition_ids = [
+            perform_substitutions(context, id)
+            for id in self.__transition_ids]
+        transition_ids = []
+        for id in subs_transition_ids:
+            transition_ids.append(int(id))
+
+        emit_actions = OrderedDict()
```

Comment:
Seems like a regular dict will do:
```suggestion
        emit_actions = {}
```